src/dune_rules/super_context.ml

```diff
@@ -270,6 +270,19 @@ let ocaml_flags t ~dir (spec : Ocaml_flags.Spec.t) =
     with_vendored_flags ~ocaml_version flags
   | false -> flags
 
+let ocaml_flags_with_melange t ~dir melange =
```

Comment:
Can you inline this function into Melange_rules?

This might require making some Super_context functions public. That is fine.

Reply:
Done in https://github.com/ocaml/dune/pull/6569/commits/83f8db78e0bd18b927242515ff02f0e2aff8a234.